wellcomeml/ml/voting_classifier.py

```diff
@@ -29,30 +30,59 @@ def _get_estimators(self):
         else:  # tuple with named estimators
             return [est for _, est in self.estimators]
 
+    def _check_vectorizer(self, estimators):
+        if type(estimators[0]) == tuple:
+            return True
+        else:
+            return False
+
     def predict(self, X):
         if self.pretrained:
-            check_is_fitted(self, 'estimators')
+            check_is_fitted(self, "estimators")
 
             estimators = self._get_estimators()
+            has_vectorizers = self._check_vectorizer(estimators)
 
             if self.voting == "soft":
-                Y_probs = np.array([est.predict_proba(X) for est in estimators])
+                if has_vectorizers:
```

Comment:
why do you add that? I am under the impression it is not necessary. an estimator can be a classifier or a pipeline that contains a vectorizer and the code should work i.e. when you call predict_proba it runs transform and then predict